### PR TITLE
Feat/nested views

### PR DESCRIPTION
**Pull request by Ascarbek**
### Feature Preview
https://www.loom.com/share/1aad74e6c5724f438f11d192030519b5

### This PR fixes:
[fixes](https://github.com/AppFlowy-IO/AppFlowy/issues/2742)

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+      name: 'New Document 1',
+      layoutType: ViewLayoutPB.Document,
+      parentViewId: page.id,
+    });
+    if (newViewResult.ok) {
+      try {
+        const newView = newViewResult.val;
+        const c = new DocumentController(newView.id);
+        await c.create();
+        await c.dispose();
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts:164]: This code needs to be deleted, it has been deleted on the main branch

File: frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+          id: newView.id,
+          showPagesInside: false,
+        })
+      );
+
+      navigate(`/page/board/${newView.id}`);
+    }
+  };
+
+  const onAddNewGridPage = async () => {
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts:215]: Maybe `onAddNewGridPage` and `onAddNewBoardPage` and `onAddNewDocumentPage` should be a `onAddPage`. The arguments are `type` `backUrl` etc.

File: frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NewFolderButton.tsx [hunk 3]
```diff
@@ -17,7 +17,7 @@ export const NewFolderButton = ({ scrollDown }: { scrollDown: () => void }) => {
           <AddSvg></AddSvg>
         </div>
       </div>
-      <span>New Folder</span>
+      <span>New View</span>
```
Comment (qinluhe) [inline frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NewFolderButton.tsx:20]: Maybe the filename should be NewViewButton and NewViewButton.hooks.ts

File: frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts (hunk 1 above)
Comment (Ascarbek) [reply frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts]: done

File: frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts (hunk 2 above)
Comment (Ascarbek) [reply frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts]: agreed

File: frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NewFolderButton.tsx (hunk 3 above)
Comment (Ascarbek) [reply frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NewFolderButton.tsx:20]: done

**Pull request merged**